sdk/python/feast/permissions/client/oidc_authentication_client_manager.py

```diff
@@ -0,0 +1,42 @@
+import requests
+
+from feast.permissions.auth_model import OidcAuthConfig
+from feast.permissions.client.auth_client_manager import AuthenticationClientManager
+
+
+class OidcAuthClientManager(AuthenticationClientManager):
+    def __init__(self, auth_config: OidcAuthConfig):
+        self.auth_config = auth_config
+
+    def _get_token_endpoint(self):
+        response = requests.get(self.auth_config.auth_discovery_url)
+        if response.status_code == 200:
+            oidc_config = response.json()
+            if not oidc_config["token_endpoint"]:
+                raise RuntimeError(
+                    " OIDC token_endpoint is not available from discovery url response."
+                )
+            return oidc_config["token_endpoint"]
+        else:
+            raise RuntimeError(
+                f"Error fetching OIDC token endpoint configuration: {response.status_code} - {response.text}"
+            )
+
+    def get_token(self):
+        # Fetch the token endpoint from the discovery URL
+        token_endpoint = self._get_token_endpoint()
+
+        token_request_body = {
+            "grant_type": "password",
+            "client_id": "admin-cli",
```

Comment:
pls use `client_id` from auth config instead (and make it mandatory instead of optional)